protos/format.proto

```diff
@@ -91,6 +91,9 @@ message IndexMetadata {
 
   // Index name. Must be unique within one dataset version.
   string name = 3;
+
+  // The max fragment_id included in this index
+  uint64 max_fragment_id = 6;
```

Comment:
Wonder should we use the version id where this index is created?
It adds one extra I/O I guess, but I feel it is more reliable , and separate of the concern where the fragment id is managed solely by the manifest.

In rare case where sometime we will reclaim fragment id, this approach might be more reliable?

Reply:
fixed